### PR TITLE
Feature #190

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

File: beam/core/src/main/java/cz/o2/proxima/beam/direct/io/CommitLogRead.java [hunk 1]
```diff
@@ -241,57 +437,56 @@ private void startObserve(String name, Partition partition, OffsetRange restrict
...
+      runningObserves = new HashMap<>();
+      partitionToSeekedOffset = new HashMap<>();
+      observers = new HashMap<>();
     }
 
-    @SplitRestriction
-    public void splitRestriction(
-        @Restriction OffsetRange restriction, OutputReceiver<OffsetRange> splits) {
+    public void tearDown() {
+      Lists.newArrayList(observers.keySet()).forEach(p -> closeHandle(p, true));
```
Comment (dmvk) [inline beam/core/src/main/java/cz/o2/proxima/beam/direct/io/CommitLogRead.java:464]: Set has a `forEach` method as well?

**Review by dmvk — APPROVED** [review]
LGTM 🎉

File: beam/core/src/main/java/cz/o2/proxima/beam/direct/io/CommitLogRead.java (hunk 1 above)
Comment (je-ik) [reply beam/core/src/main/java/cz/o2/proxima/beam/direct/io/CommitLogRead.java:464]: Hm, no idea why it previously cloned the key set. I suppose that was part of some debugging.

Comment (je-ik) [reply beam/core/src/main/java/cz/o2/proxima/beam/direct/io/CommitLogRead.java:464]: I have idea now. The `closeHandle` removes the observer from the `observers`, so not cloning the key set results in ConcurrentModificationException.

**Comment by je-ik** [comment]
This closes O2-Czech-Republic#190

**Pull request merged**